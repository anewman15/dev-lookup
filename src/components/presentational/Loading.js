import common from '../../styles/commonStyles.module.css';

const Loading = () => (
  <div className={`${common.container} ${common.flexCenter} ${common.m1}`}>
    <img src={`${process.env.PUBLIC_URL}/loading.gif`} alt="loading" />
  </div>
);

export default Loading;
