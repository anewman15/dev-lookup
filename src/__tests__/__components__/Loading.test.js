import { unmountComponentAtNode } from 'react-dom';
import renderer from 'react-test-renderer';
import Loading from '../../components/presentational/Loading';

let container = null;
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe('Loading', () => {
  it('renders a loading gif spinner', () => {
    const tree = renderer
      .create(<Loading />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
