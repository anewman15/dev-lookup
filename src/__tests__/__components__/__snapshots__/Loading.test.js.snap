// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading renders a loading gif spinner 1`] = `
<div
  className="container flexCenter m1"
>
  <img
    alt="loading"
    src="/loading.gif"
  />
</div>
`;
